pkg/pillar/cmd/zedmanager/handleresolve.go

```diff
@@ -0,0 +1,183 @@
+	config := lookupResolveConfig(ctx, status.Key())
+	if config != nil {
+		log.Infof("handleResolveStatusDelete delete config for %s\n",
+			key)
+		unpublishResolveConfig(ctx, config)
+	}
+	log.Infof("handleResolveStatusDelete done for %s\n", key)
+}
+
+func handleResolveStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleResolveStatusModify for %s\n", key)
+	ctx := ctxArg.(*zedmanagerContext)
+	rs := statusArg.(types.ResolveStatus)
+	pub := ctx.pubAppInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+		config := lookupAppInstanceConfig(ctx, status.Key())
+		if config == nil {
+			errStr := fmt.Sprintf("App Instance config not found for %s while resolving tags\n",
+				status.Key())
+			log.Error(errStr)
+			status.SetError(errStr, time.Now())
+			publishAppInstanceStatus(ctx, &status)
+			continue
+		}
+		for i := range status.StorageStatusList {
+			ss := &status.StorageStatusList[i]
+			if !ss.HasResolverRef || ss.Name != rs.Name || ss.DatastoreID != rs.DatastoreID {
+				continue
+			}
+			log.Infof("Updating SHA of storage status (%v) in app instance (%v)\n",
+				ss.ImageID, status.UUIDandVersion.UUID)
+			if len(rs.Error) != 0 {
+				errStr := fmt.Sprintf("Error occurred while resolving tags, updating app instance %s.\n",
+					status.Key())
+				log.Error(errStr)
+				status.SetErrorWithSource(rs.Error, types.ResolveStatus{}, rs.ErrorTime)
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			if rs.ImageSha256 == "" {
+				errStr := fmt.Sprintf("Empty SHA in resolve status, updating app instance %s.\n",
+					status.Key())
+				log.Error(errStr)
+				status.SetError(errStr, time.Now())
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			sc := lookupStorageConfig(config, *ss)
+			if sc == nil {
+				errStr := fmt.Sprintf("Storage config (%v) not found for app instance (%s) while resolving tags\n",
+					ss.ImageID, status.Key())
+				log.Error(errStr)
+				status.SetError(errStr, time.Now())
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			if ss.ImageSha256 == "" {
+				log.Infof("Image SHA (%s) found while resolving status for storage status (%s)\n",
+					rs.ImageSha256, ss.ImageID)
+				ss.ImageSha256 = rs.ImageSha256
+			} else if ss.ImageSha256 != rs.ImageSha256 {
+				log.Infof("Image SHA changed from (%s) to (%s) while resolving status for storage status (%s)\n",
+					ss.ImageSha256, rs.ImageSha256, ss.ImageID)
+				MaybeRemoveStorageStatus(ctx, status.UUIDandVersion.UUID, ss)
+				deleteAppAndImageHash(ctx, status.UUIDandVersion.UUID,
+					ss.ImageID)
+				ss.UpdateFromStorageConfig(*sc)
+				ss.ImageSha256 = rs.ImageSha256
+			} else {
+				log.Infof("Image SHA (%s) not changed for storage status (%s)\n",
+					ss.ImageSha256, ss.ImageID)
+			}
+			addAppAndImageHash(ctx, config.UUIDandVersion.UUID,
+				ss.ImageID, ss.ImageSha256)
+			rc := lookupResolveConfig(ctx, rs.Key())
```

Comment:
As a stylistic thing this should be gated by checking 
 if ss.HasResolverRef (because that is checked 50 lines above so not in the local context of the reader)
and then it should clear ss.HasResolverRef

You might want to add a small DeleteResolveConfig() which then does the lookupResolveConfig plus unpublish which is what we have for the other services.

Reply:
Resolved